timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -254,7 +286,93 @@ def _fit(
         **kwargs,
     ) -> None:
         self._check_fit_params()
-        self.time_limit = time_limit
+
+        fine_tune_time_limit = time_limit // 2 if self.fine_tune else 0
+        self.time_limit = time_limit - fine_tune_time_limit  # inference time budget
+
+        if self.fine_tune:
+            context_length = self._get_context_length(train_data)
+            # load model pipeline to device memory
+            self.load_model_pipeline(context_length=context_length, is_training=True)
+
+            output_dir = Path(self.path) / "transformers_logs"
+            training_args = TrainingArguments(
+                output_dir=str(output_dir),
+                per_device_train_batch_size=self.fine_tune_batch_size,
+                learning_rate=self.fine_tune_lr,
+                lr_scheduler_type="linear",
+                warmup_ratio=0.0,
+                optim="adamw_torch_fused",
+                logging_dir=str(output_dir),
+                logging_strategy="steps",
+                logging_steps=self.logging_steps,
+                save_strategy="steps",
+                save_steps=self.eval_and_save_steps,
+                evaluation_strategy="steps",
+                eval_steps=self.eval_and_save_steps,
+                report_to="none",
+                max_steps=self.fine_tune_steps,
+                gradient_accumulation_steps=1,
+                dataloader_num_workers=self.data_loader_num_workers,
+                tf32=self._has_tf32(),
+                save_only_model=True,
+                load_best_model_at_end=True,
+                metric_for_best_model="eval_loss",
+                prediction_loss_only=True,
+                save_total_limit=1,
+            )
+            tokenizer_train_dataset = ChronosFineTuningDataset(
+                tokenizer=self.model_pipeline.tokenizer,
+                target_df=train_data,
+                freq=self.freq,
+                target_column=self.target,
+                context_length=context_length,
+                prediction_length=self.prediction_length,
+                mode="training",
+            )
+            callbacks = [TimeLimitCallback(time_limit=fine_tune_time_limit), EvaluateAndSaveFinalStepCallback()]
+
+            tokenizer_val_dataset = (
```

Comment:
why are we using the validation data? in order to do early stopping or only for selecting checkpoints? If we're not doing early-stopping I think we can skip this entirely to prevent val data from leaking into the training here (the same data will be used to fit the ensemble).

Reply:
We are selecting the best checkpoint based on validation. I discussed this with @shchur and we decided that we will have an option to perform such validation but will turn in off by default.

Reply:
I kept the option to perform validation but turned it off by default.